lib/rex/proto/mssql/client_mixin.rb

```diff
@@ -86,6 +86,92 @@ def mssql_print_reply(info)
     end
   end
 
+  def mssql_prelogin_packet
+    pkt = ""
+    pkt_hdr = ""
+    pkt_data_token = ""
+    pkt_data = ""
+
+
+    pkt_hdr = [
+        TYPE_PRE_LOGIN_MESSAGE, #type
+        STATUS_END_OF_MESSAGE, #status
+        0x0000, #length
+        0x0000, # SPID
+        0x00, # PacketID
+        0x00 #Window
+    ]
+
+    version = [0x55010008, 0x0000].pack("Vv")
+
+    # if manually set, we will honour
+    if tdsencryption == true
+      encryption = ENCRYPT_ON
+    else
+      encryption = ENCRYPT_NOT_SUP
+    end
+
+    instoptdata = "MSSQLServer\0"
+
+    threadid = "\0\0" + Rex::Text.rand_text(2)
+
+    idx = 21 # size of pkt_data_token
+    pkt_data_token << [
+        0x00, # Token 0 type Version
+        idx , # VersionOffset
+        version.length, # VersionLength
+
+        0x01, # Token 1 type Encryption
+        idx = idx + version.length, # EncryptionOffset
+        0x01, # EncryptionLength
+
+        0x02, # Token 2 type InstOpt
+        idx = idx + 1, # InstOptOffset
+        instoptdata.length, # InstOptLength
+
+        0x03, # Token 3 type Threadid
+        idx + instoptdata.length, # ThreadIdOffset
+        0x04, # ThreadIdLength
+
+        0xFF
+    ].pack("CnnCnnCnnCnnC")
+
+    pkt_data << pkt_data_token
+    pkt_data << version
+    pkt_data << encryption
+    pkt_data << instoptdata
+    pkt_data << threadid
+
+    pkt_hdr[2] = pkt_data.length + 8
+
+    pkt = pkt_hdr.pack("CCnnCC") + pkt_data
+    pkt
+  end
+
+  def mssql_get_version
```

Comment:
Instead of this specific method; it might be a better long-term implementation to just have this method return a hash of the parsed values that the server has returned

i.e. the version number, and whether encryption is supported - and maybe in the future it would have more details

Reply:
Potentially it could return all of this parsed-metadata for the client to decide what they want to extract

![image](https://github.com/rapid7/metasploit-framework/assets/60357436/1e9f4e65-d4c6-4d94-a919-30ee5e735b4d)

Reply:
Looks like we had the same thought, I included this in the last commit